TESTS/events/equeue/main.cpp

```diff
@@ -0,0 +1,832 @@
+        timing->delay = (i + 1) * 100;
+        equeue_event_delay(timing, timing->delay);
+        equeue_event_period(timing, timing->delay);
+
+        int id = equeue_post(&q, timing_func, timing);
+        TEST_ASSERT(id);
+    }
+
+    equeue_dispatch(&q, N * 100);
+
+    equeue_destroy(&q);
+}
+
+
+template<int N>
+static void fragmenting_barrage_test()
+{
+    equeue_t q;
+    int err = equeue_create(&q,
+                            2 * N * (EQUEUE_EVENT_SIZE + sizeof(struct fragment) + N * sizeof(int)));
+    TEST_ASSERT(!err);
+
+    for (int i = 0; i < N; i++) {
+        size_t size = sizeof(struct fragment) + i * sizeof(int);
+        struct fragment *fragment = reinterpret_cast<struct fragment *>(equeue_alloc(&q, size));
+        TEST_ASSERT(fragment);
+
+        fragment->q = &q;
+        fragment->size = size;
+        fragment->timing.tick = equeue_tick();
+        fragment->timing.delay = (i + 1) * 100;
+        equeue_event_delay(fragment, fragment->timing.delay);
+
+        int id = equeue_post(&q, fragment_func, fragment);
+        TEST_ASSERT(id);
+    }
+
+    equeue_dispatch(&q, N * 100);
+
+    equeue_destroy(&q);
+}
+
+struct ethread {
+    Thread thread;
+    equeue_t *q;
+    int ms;
+};
+
+static void ethread_dispatch(void *p)
+{
+    struct ethread *t = (struct ethread *)p;
+    equeue_dispatch(t->q, t->ms);
+}
+
+template<int N>
+static void multithreaded_barrage_test()
+{
+    equeue_t q;
+    int err = equeue_create(&q, N * (EQUEUE_EVENT_SIZE + sizeof(struct timing)));
+    TEST_ASSERT(!err);
+
+    struct ethread t;
+    t.q = &q;
+    t.ms = N * 100;
+
+    t.thread.start(callback(ethread_dispatch, &t));
+
+    for (int i = 0; i < N; i++) {
+        struct timing *timing = reinterpret_cast<struct timing *>(equeue_alloc(&q, sizeof(struct timing)));
+        TEST_ASSERT(timing);
+
+        timing->tick = equeue_tick();
+        timing->delay = (i + 1) * 100;
+        equeue_event_delay(timing, timing->delay);
+        equeue_event_period(timing, timing->delay);
+
+        int id = equeue_post(&q, timing_func, timing);
+        TEST_ASSERT(id);
+    }
+
+    err = t.thread.join();
+    TEST_ASSERT(!err);
+
+    equeue_destroy(&q);
+}
+
+struct count_and_queue {
+    int p;
+    equeue_t *q;
+};
+
+void simple_breaker(void *p)
+{
+    struct count_and_queue *caq = (struct count_and_queue *)p;
+    equeue_break(caq->q);
+    ThisThread::sleep_for(10);
+    caq->p++;
+}
+
+static void break_request_cleared_on_timeout_test()
+{
+    equeue_t q;
+    int err = equeue_create(&q, 2048);
+    TEST_ASSERT(!err);
+
+    struct count_and_queue pq;
+    pq.p = 0;
+    pq.q = &q;
+
+    int id = equeue_call_every(&q, 10, simple_breaker, &pq);
+
+    equeue_dispatch(&q, 10);
+    TEST_ASSERT(pq.p == 1);
+
+    equeue_cancel(&q, id);
+
+    int count = 0;
+    equeue_call_every(&q, 10, simple_func, &count);
+
+    equeue_dispatch(&q, 55);
+    TEST_ASSERT(count > 1);
+
+    equeue_destroy(&q);
+}
+
+void sibling_test(void)
```

Comment:
Add description in `Given/When/Then` format to every test case function, describing test scenario.
Example below:
```
/** Test that ::rtc_init can be called multiple times.
 *
 *  Given board provides RTC.
 *  When ::rtc_init is called multiple times.
 *  Then ::rtc_init are successfully performed (no exception is generated).
 *
 */
```